actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -18,6 +18,16 @@ module ActionDispatch
   class HostAuthorization
     ALLOWED_HOSTS_IN_DEVELOPMENT = [".localhost", IPAddr.new("0.0.0.0/0"), IPAddr.new("::/0")]
     PORT_REGEX = /(?::\d+)?/.freeze
+    IPV6_HOSTNAME = /(?<host>[a-f0-9]*:[a-f0-9.:]+)/i
+    IPV4_HOSTNAME = /(?<host>\d+.\d+.\d+.\d+)/i
+    IPV4_HOSTNAME_WITH_PORT = /\A#{IPV4_HOSTNAME}#{PORT_REGEX}?\z/
+    IPV6_HOSTNAME_WITH_PORT = /\A\[#{IPV6_HOSTNAME}\]#{PORT_REGEX}?\z/
+    VALID_IP_HOSTNAME = Regexp.union(
+      /\A#{IPV4_HOSTNAME}\z/,
+      /\A#{IPV6_HOSTNAME}\z/,
+      IPV4_HOSTNAME_WITH_PORT,
+      IPV6_HOSTNAME_WITH_PORT
```

Comment:
Should we make the `\A` / `\z` anchors consistent?

Reply:
Yes. I can move them to the union call, but all those regexp already have the same anchors.